pkg/controller/logstorage/users/user_controller.go

```diff
@@ -0,0 +1,261 @@
+		}
+	}
+
+	return nil
+}
+
+func (r *UserController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling LogStorage - Users")
+
+	var tenant *operatorv1.Tenant
+	var err error
+	if r.multiTenant {
+		if request.Namespace == "" {
+			// In multi-tenant mode, we only handle namespaced reconcile triggers.
+			return reconcile.Result{}, nil
+		}
+
+		// Check if there is a manager in this namespace.
+		tenant, err = utils.GetTenant(ctx, r.client, request.Namespace)
+		if errors.IsNotFound(err) {
+			// No tenant in this namespace. Ignore the update.
+			reqLogger.Info("No Tenant in this Namespace, skip")
+			return reconcile.Result{}, nil
+		} else if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred while querying Tenant", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Get LogStorage resource.
+	ls := &operatorv1.LogStorage{}
+	key := utils.DefaultTSEEInstanceKey
+	err = r.client.Get(ctx, key, ls)
+	if err != nil {
+		// Not finding the LogStorage CR is not an error, as a Managed cluster will not have this CR available but
+		// there are still "LogStorage" related items that need to be set up
+		if !errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "An error occurred while querying LogStorage", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		ls = nil
+		r.status.OnCRNotFound()
+	}
+
+	// We found the LogStorage instance (and Tenant instance if in multi-tenant mode).
+	r.status.OnCRFound()
+
+	req := octrl.NewRequest(request.NamespacedName, r.multiTenant, render.ElasticsearchNamespace)
+	args := ReconcileArgs{LogStorage: ls, Tenant: tenant}
+	reqLogger = reqLogger.WithValues("installNS", req.InstallNamespace(), "truthNS", req.TruthNamespace())
+	return r.reconcile(ctx, reqLogger, args, req)
+}
+
+type ReconcileArgs struct {
+	LogStorage *operatorv1.LogStorage
+	Tenant     *operatorv1.Tenant
+}
+
+func (r *UserController) reconcile(ctx context.Context, reqLogger logr.Logger, args ReconcileArgs, request octrl.Request) (reconcile.Result, error) {
+	// Wait for the initializing controller to indicate that the LogStorage object is actionable.
+	if args.LogStorage.Status.State != operatorv1.TigeraStatusReady {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for LogStorage defaulting to occur", nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+
+	tenantID := ""
+	if args.Tenant != nil {
+		tenantID = args.Tenant.Spec.ID
+		if tenantID == "" {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Tenant resource does not specify an ID", nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+	}
+	linseedUser := utils.LinseedUser(tenantID)
+
+	// Query any existing username and password for this Linseed instance. If one already exists, we'll simply
```

Comment:
I think we want a way to specify manually provided user credentials so that they can be provisioned out of band.

we'll also want a way for this controller to clean up credentials that are no longer needed, probably.